### DISCUSSION TITLE
[Bug]: Messages coming through incomplete, filled with '�' and other symbols

### DISCUSSION THREAD

**Discussion by ConnorDFWT** [meshtastic/firmware#4993] [Troubleshooting]
### Category

Serial

### Hardware

Heltec V3, Other

### Firmware Version

2.4.3.91d6612

### Description

I've been trying to test out relaying messages from an Arduino Nano 33 BLE Sense to a Heltec V3 through serial connection, and then have that message sent out on meshtastic. However, when the message is received by another Heltec V3 client, it never displays as intended, only parts or none at all of the original message come through. 

I wrote some quick code to send a message to the Heltec every 30 seconds (at bottom of the post), which displayed on the Arduino IDE and should have come out as:

> 15:16:10.119 -> Hello from Arduino. Test: 10
> 15:16:40.147 -> Hello from Arduino. Test: 11
> 15:17:10.129 -> Hello from Arduino. Test: 12

But instead was received as:

> Henlo཯ �
> ���
> ɑե��� Test: 11
> Hello཯

My problem seemed similar to issue #3452, but I haven't found a solution that works for me there. I've tried flashing 2.5 and 2.3 firmware, changing the rx/tx pins, changing baud rates and toggling almost all of the settings I could think of. I've attached some screenshots of the debug log. Thank you all so much!

![1000005593](https://github.com/user-attachments/assets/e08a4ce0-b872-488e-bee8-68faa4369f23)
![1000005590](https://github.com/user-attachments/assets/6eb54142-ce6a-4400-bbd8-a89c809291ae)

_Arduino Code:_
```
int counter;

void setup() {
  Serial.begin(9600);
  Serial1.begin(115200);
}

void loop() {
  if (Serial1.available()) {
    Serial.print("Hello from Arduino. Test: ");
    Serial.println(counter);
    Serial1.print("Hello from Arduino. Test: ");
    Serial1.println(counter);
    counter ++;
    delay (30000);
    
  }
  else {
    Serial.println("Serial not connected");
    delay (300);
  }
}
```

**Comment by thebentern** [comment]
Looks like a character encoding issue?
